app/controllers/assignments_controller.rb

```diff
@@ -577,74 +582,76 @@ def download_config_files
   # Uploads a zip file containing all the files specified in download_config_files
   # and modifies the assignment settings according to those files.
   def upload_config_files
-    error = false
     upload_file = params.require(:upload_files_for_config)
-    if upload_file.size == 0
-      error = true
-      flash_message(:error, I18n.t('upload_errors.blank'))
-    end
-    if File.extname(upload_file.path).casecmp?('.zip') && !error
-      Zip::File.open(upload_file.path) do |zipfile|
-        Assignment.transaction do
-          # Build assignment from properties
-          prop_file = zipfile.find_entry('properties.yml')
-          assignment = build_uploaded_assignment(prop_file)
-          zipfile.remove(prop_file)
-          # Build peer review assignment if it exists
-          child_prop_file = zipfile.find_entry('peer-review-config-files/properties.yml')
-          unless child_prop_file.nil?
-            child_assignment = build_uploaded_assignment(child_prop_file, assignment)
-            child_assignment.save!
-            zipfile.remove(child_prop_file)
-          end
-          zipfile.each do |entry|
-            flash_message(:warning, I18n.t('assignments.unexpected_file_found', item: entry.name))
-          end
-          assignment.save!
-          redirect_to edit_assignment_path(assignment.id)
-        rescue StandardError => e
-          error = true
-          flash_message(:error, e.message)
-          raise ActiveRecord::Rollback
+    raise I18n.t('upload_errors.blank') if upload_file.size == 0
```

Comment:
If it fits on a single line, it is preferable to use a guard clause like these ones